nengo/builder/processes.py

```diff
@@ -98,7 +102,7 @@ def make_step(self, signals, dt, rng):
         shape_in = input.shape if input is not None else (0,)
         shape_out = output.shape
         rng = self.process.get_rng(rng)
-        state = {name: signals[sig] for name, sig in self.state.items()}
+        state = {name: signals[self.updates[idx]] for name, idx in self.state.items()}
```

Comment:
Do we need this for the optimizer to work on processes? If so, what was the optimizer doing before? Should we add a test that the optimizer works properly with processes?

Reply:
I believe right now the optimizer does not work on processes, you can't merge two processes into one. However, it might be possible for a signal involved in the process to be merged as a result of merging some other op. My guess is that it doesn't actually occur for process state variables, even though it did occur for neuron type state variables (which is where I encountered the issue, and spurred me to make the same changes for process just in case it comes up in the future). I'll see if I can think of a way to test it, but probably it'll take more effort than it's worth.

Reply:
Yeah, not worth putting time into. Makes sense to have this for the future.